src/base/RegexPattern.cc

```diff
@@ -8,34 +8,29 @@
 
 #include "squid.h"
 #include "base/RegexPattern.h"
-#include "base/TextException.h"
 #include "debug/Stream.h"
 #include "sbuf/Stream.h"
 
 #include <iostream>
-#include <utility>
 
-RegexPattern::RegexPattern(const SBuf &aPattern, const int aFlags):
+RegexPattern::RegexPattern(const SBuf &aPattern, const std::regex::flag_type aFlags):
     pattern(aPattern),
-    flags(aFlags)
+    regex(aPattern.rawContent(), aPattern.length(), aFlags)
 {
-    memset(&regex, 0, sizeof(regex)); // paranoid; POSIX does not require this
-    if (const auto errCode = regcomp(&regex, pattern.c_str(), flags)) {
```

Comment:
Have you tried to measure the performance impact of us switching to std::regex? I understand that the performance depends on many variables, but it would be good to confirm that we are not slowing down Squids a lot by this upgrade in a typical/popular environment with some basic regexes (at least).

Reply:
Have now done a performance check with a test program (not squid itself) - performing regex search/match 100K times to match a URL. Inefficient, but the pattern is one admin seems to like using.

regcomp() ...  35ms +/- 1ms.
std::regex  ...  480-500ms +/- 10ms

Reply:
For the heck of it I tried the other regex syntax and they all turned out worse performance by 30-200ms than POSIX ERE.

Reply:
Doh. test did not have optimizations involved. Simply adding `-O2` we get far better numbers

regcomp() ... 35ms +/- 1ms.
std::regex ... 98ms +/- 8ms